frontend/src/product/pages/Cart.js

```diff
@@ -0,0 +1,30 @@
+import React,{useState,} from 'react';
+import { Card, Button, Modal  } from 'semantic-ui-react'
+const Cart = ({ products,button }) => {
+    const [open, setOpen] =useState(false)
+    return ( <Modal
+            onClose={() => setOpen(false)}
+            onOpen={() => setOpen(true)}
+            open={open}
+             trigger={button}
+        >
+            <Modal.Header>YOUR SHOPPING CART</Modal.Header>  
+            <Modal.Content>
+          {products.length !==0 ?  products.map(item =>
+                    <Card centered
+                        image={`https://res.cloudinary.com/dfdocrzsc/image/upload/v1607554931/supermarket/${item.filename}`}
+                        header={`${item.title}`}
+                        meta={ `$ ${item.price}`}
+                        description={`${item.description}`}
+                    
+                    />): <h3>YOUR CART IS EMPTY</h3>}
```

Comment:
Nice edge case handling.
Only thing is that the message should not be tagged with h3 since it's not a header.